packages/relay/src/lib/eth.ts

```diff
@@ -665,15 +665,15 @@ export class EthImpl implements Eth {
                 }
 
                 balanceFound = true;
-                weibars = (currentBalance - balanceFromTxs) * constants.TINYBAR_TO_WEIBAR_COEF;
+                weibars = BigInt(currentBalance - balanceFromTxs) * BigInt(constants.TINYBAR_TO_WEIBAR_COEF);
```

Comment:
Q: If a lot of these numbers were just BigInt by type definition would we need all the casting?
Or is this an optimization of sort?

Reply:
No optimization in mind. I haven't defined them as BigInt, because in some other places where we are using those variables and constants, they are needed as numbers and that means converting them back to number. Problem is that when we multiply them by the weibar coefficient, we are losing the precision, when they are numbers or bignumbers. I've cached this coefficient as bigInt at the beginning of the tests to avoid casting, but for the other numbers I can't avoid the casting either from BigInt to number or the other way around.